scout/build/panel.py

```diff
@@ -1,94 +1,61 @@
-        )
+    hgnc_id = gene_info.get("hgnc_id")
+
+    # Validate presence of hgnc_id
+    if not hgnc_id:
+        raise KeyError(f"Gene {symbol} is missing hgnc id. Panel genes must have hgnc_id.")
 
+    # Try fetching gene information from either build "37" or "38"
     hgnc_gene = adapter.hgnc_gene_caption(
         hgnc_identifier=hgnc_id, build="37"
     ) or adapter.hgnc_gene_caption(hgnc_identifier=hgnc_id, build="38")
+
     if hgnc_gene is None:
-        raise IntegrityError("hgnc_id {0} is not in the gene database!".format(hgnc_id))
+        raise IntegrityError(f"hgnc_id {hgnc_id} is not in the gene database!")
 
-    gene_obj = dict(hgnc_id=hgnc_id)
+    gene_obj = {"hgnc_id": hgnc_id, "symbol": hgnc_gene["hgnc_symbol"]}
 
-    gene_obj["symbol"] = hgnc_gene["hgnc_symbol"]
+    # Log warnings if symbols do not match
     if symbol != gene_obj["symbol"]:
-        LOG.warning(
-            "Symbol in database does not correspond to symbol in panel file for gene %s",
-            hgnc_id,
-        )
-        LOG.warning(
-            "Using symbol %s for gene %s, instead of %s"
-            % (hgnc_gene["hgnc_symbol"], hgnc_id, symbol)
-        )
+        LOG.warning(f"Symbol in database does not match symbol in panel file for gene {hgnc_id}")
+        LOG.warning(f"Using symbol {gene_obj['symbol']} for gene {hgnc_id} instead of {symbol}")
+
+    # Add optional gene information
+    gene_obj.update(
+        {
+            key: gene_info[key]
+            for key in ["disease_associated_transcripts", "comment", "database_entry_version"]
+            if key in gene_info
+        }
+    )
+
+    # Add boolean flags
+    gene_obj.update(
+        {key: True for key in ["reduced_penetrance", "mosaicism"] if gene_info.get(key)}
+    )
+
+    # Handle inheritance models
+    if "inheritance_models" in gene_info:
+        gene_obj["inheritance_models"] = gene_info["inheritance_models"]
 
-    if gene_info.get("transcripts"):
-        gene_obj["disease_associated_transcripts"] = gene_info["transcripts"]
-
-    for gene_member_variable_str in ["comment", "database_entry_version"]:
-        if gene_info.get(gene_member_variable_str):
-            gene_obj[gene_member_variable_str] = gene_info[gene_member_variable_str]
-
-    for gene_member_variable_bool in ["reduced_penetrance", "mosaicism"]:
-        if gene_info.get(gene_member_variable_bool):
-            gene_obj[gene_member_variable_bool] = True
-
-    if gene_info.get("inheritance_models"):
-        gene_obj["inheritance_models"] = []
-        custom_models = []
-        for model in gene_info["inheritance_models"]:
-            if model not in VALID_MODELS:
-                custom_models.append(model)
-                continue
-            gene_obj["inheritance_models"].append(model)
-            lc_model = model.lower()  # example ad = True
-            gene_obj[lc_model] = True
-        gene_obj["custom_inheritance_models"] = custom_models
```

Comment:
This was not working at all, since the custom_inheritance_models (can be manually set as a comma separated list ) was set to the inheritance_models that are the preset multiselect instead.